Flaky test: GrizzlyLDAPConnectionFactoryTestCase.testResourceManagement times out on macOS/Java 11

### Summary

`GrizzlyLDAPConnectionFactoryTestCase.testResourceManagement` is a flaky test that intermittently fails by timeout, breaking otherwise-green CI builds.

Observed in CI run [#29592628645](https://github.com/OpenIdentityPlatform/OpenDJ/actions/runs/29592628645), job [`build-maven (macos-latest, 11)`](https://github.com/OpenIdentityPlatform/OpenDJ/actions/runs/29592628645/job/87925212120). 8 of 9 `build-maven` matrix jobs passed (all Ubuntu 11/17/21/25/26, Windows 11/26, macOS 26); only **macOS + Java 11** failed.

### Failure

```
[ERROR] org.forgerock.opendj.grizzly.GrizzlyLDAPConnectionFactoryTestCase.testResourceManagement -- Time elapsed: 30.08 s <<< FAILURE!
[ERROR]   GrizzlyLDAPConnectionFactoryTestCase.testResourceManagement:333 expected:<true> but was:<false>
[ERROR] Tests run: 1038, Failures: 1, Errors: 0, Skipped: 0
```

### Root cause

The test (`opendj-grizzly/src/test/java/org/forgerock/opendj/grizzly/GrizzlyLDAPConnectionFactoryTestCase.java:320`, regression test for OPENDJ-1156) runs **100 iterations**; each opens a connection, triggers a remote `disconnect()`, and waits up to `TEST_TIMEOUT = 30s` for the client-side error notification:

```java
listener.awaitError(TEST_TIMEOUT, TimeUnit.SECONDS);   // line 333
```

which ultimately asserts in `MockConnectionEventListener.await()` (`opendj-core/src/test/java/org/forgerock/opendj/ldap/MockConnectionEventListener.java:84`):

```java
assertThat(errorLatch.await(timeout, unit)).isTrue();
```

The assertion returned `false`, i.e. on one of the 100 iterations the `errorLatch` was never released — the connection-error notification did not reach the client within 30 s (elapsed time 30.08 s confirms the timeout was hit). This is timing/environment-sensitive: it reproduced only on the macOS/Java 11 runner and passed on every other OS/JDK combination.

### Impact

Not a functional regression and unrelated to the PR under test (Windows distribution / MSI). It is CI flakiness that produces false-negative builds and forces re-runs.

### Suggested fix

Stabilize the async test, e.g.:
- reduce the iteration count for CI, and/or
- make the wait more robust (adaptive/longer timeout, or assert on the notification rather than a hard latch deadline).

### Workaround

Re-running the failed job typically passes:

```
gh run rerun --repo OpenIdentityPlatform/OpenDJ 29592628645 --failed
```

--- a/opendj-grizzly/src/test/java/org/forgerock/opendj/grizzly/GrizzlyLDAPConnectionFactoryTestCase.java
+++ b/opendj-grizzly/src/test/java/org/forgerock/opendj/grizzly/GrizzlyLDAPConnectionFactoryTestCase.java
@@ -12,6 +12,7 @@
  * information: "Portions Copyright [year] [name of copyright owner]".
  *
  * Copyright 2013-2016 ForgeRock AS.
+ * Portions Copyrighted 2026 3A Systems, LLC
  */
 package org.forgerock.opendj.grizzly;
 
@@ -86,8 +87,16 @@ public class GrizzlyLDAPConnectionFactoryTestCase extends SdkTestCase {
      */
     private static final int ITERATIONS = 100;
 
-    /** Test timeout for tests which need to wait for network events. */
-    private static final long TEST_TIMEOUT = 30L;
+    /**
+     * Test timeout for tests which need to wait for network events. This is a
+     * generous upper bound rather than an expected duration: in the normal case
+     * the awaited latch/promise is released within milliseconds. It only takes
+     * effect when a CI runner is slow enough to delay delivery of an async
+     * event (see issue #748, where {@code testResourceManagement} timed out on
+     * a loaded macOS/Java 11 runner), so a larger value trades a slower failure
+     * for far fewer false-negative builds.
+     */
+    private static final long TEST_TIMEOUT = 60L;
 
     /*
      * It is usually quite a bad code smell to share state between unit tests.

